### PR TITLE
[WIP] Initial version

### PR DESCRIPTION
**Pull request by gr2m**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = App
+
+const jsonwebtoken = require('jsonwebtoken')
+
+function App ({id, privateKey}) {
+  const api = {
+    async appAuth () {
```
Comment (hiimbex) [inline index.js]: nit: do we need to make this async? it doesn't appear that we are awaiting on anything

**Comment by gr2m** [comment]
> The only difference is instead of {id, privateKey} they use options which contains `{id, privateKey}

I think that’s the same thing? Looks like Probot uses `cert` instead of `privateKey`, is that what you mean?

File: index.js (hunk 1 above)
Comment (gr2m) [reply index.js]: yeah good point!

**Comment by hiimbex** [comment]
Well it's the different of saying `id` or `options.id`, but again, not very important and probably better as is!

**Comment by hiimbex** [comment]
WHOOPs

**Comment by gr2m** [comment]
Instead of merging this I thought you can continue working on the branch and finish it up? I can also clean up the commits and merge and you take it from there, whatever you prefer

**Comment by hiimbex** [comment]
Sounds good! Next step will be getting the installation access token. Do you think that should use the octokit internally to do that or would it be better to use something like `got`?

**Comment by gr2m** [comment]
ah good point! I’d like to use https://github.com/octokit/rest.js/tree/master/lib/request, we can make this its own little package (e.g. `@octokit/request`) as well so it would be easy to use across `rest.js`, `app.js`, `graphql.js` etc and for folks to use standalone as a minimal request library for the GitHub APIs 

For now, you can add `@octokit/rest` as dependency, pin the version and then do 

```
cont request = require('@octokit/rest/lib/request')
```

**Comment by hiimbex** [comment]
That's what I was thinking! We try to use octokit/request for [early stuff](https://github.com/probot/probot/blob/76b0417745d89180c638b044e2e8a5fd2285455d/src/manifest-creation.ts#L60-L64) too.

But I think it's worth making that it's own small package separate of @octokit/rest eventually, so that this module is truly standalone!

**Comment by hiimbex** [comment]
Added the basics for installation access token and setup travis on this repo! Feel free to edit the settings or the config, I tried to just do the bare minimum. :)

File: test/test.js [hunk 2]
```diff
@@ -50,15 +50,15 @@ describe('app.js', function () {
...
+    const bearer = app.appAuth()
     expect(bearer).to.equal(BEARER)
   })
 
   // see https://developer.github.com/v3/apps/#create-a-new-installation-token
   it('gets installation token', async () => {
     nock('https://api.github.com', {
-      reqheaders: {
+      headers: {
         authorization: `bearer ${BEARER}` // installation access token
```
Comment (gr2m) [inline test/test.js]: I think this must be `reqheaders`: https://github.com/nock/nock#specifying-request-headers

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+language: node_js
+node_js:
+  - "8"
+
+install:
+  - npm install
+
+script:
+  - npm test
```
Comment (gr2m) [inline .travis.yml]: both `npm install` and `npm test` are the defaults for either configuration so we can leave them out. But we can also make it explicit if you prefer

**Comment by gr2m** [comment]
hmm when running `npm test` or directly `./node_modules/.bin/mocha` the tests seem to hang, the process never ends. Dunno why, shall I help investigate?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    async requestToken ({installationId}) {
+      const res = await request({
+        // POST /app/installations/:installation_id/access_tokens
+        method: 'POST',
+        url: `/app/installations/:installation_id/access_tokens`,
+        installation_id: installationId,
+        headers: {
+          accept: 'application/vnd.github.machine-man-preview+json'
+        },
+        // TODO: store the appAuth() if it's been elss than 60 minutes
```
Comment (gr2m) [inline index.js]:
```suggestion
        // TODO: store the appAuth() if it's been less than 60 minutes
```

^^ Just wanted to try the new suggestion feature!

File: index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        authorization: `bearer ${this.appAuth()}`
```
Comment (gr2m) [inline index.js]: `authorization` goes into the `headers` object above

**Pull request merged**